oauth_request.go

```diff
@@ -0,0 +1,53 @@
+package geddit
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+type oauthRequest struct {
+	accessToken string
+	url         string
+	useragent   string
+	values      *url.Values
+}
+
+func (r oauthRequest) getResponse() (*bytes.Buffer, error) {
+	// Determine the HTTP action.
+	var action, finalurl string
+	if r.values == nil {
+		action = "GET"
+		finalurl = r.url
+	} else {
+		action = "POST"
+		finalurl = r.url + "?" + r.values.Encode()
+	}
+
+	// Create a request and add the proper headers.
+	req, err := http.NewRequest(action, finalurl, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", r.useragent)
+	req.Header.Set("Authorization", "bearer "+r.accessToken)
+
+	// Handle the request
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Ah, thanks for clearing that up. Is this more of what you had in mind?

```go
type oauthClient struct {
       http.Client
}

func (o *oauthClient) getResponse(oReq *oauthRequest) (*bytes.Buffer, error) {
...
}
```

Then API endpoint functions would do something like...

```go
req := &oauthRequest{
        accessToken: s.accessToken,
        url:         "https://oauth.reddit.com/api/v1/me",
        useragent:   s.useragent,
}
client := new(oauthClient)
body, err := client.getResponse(req)
```